casepro/backend/junebug.py

```diff
@@ -31,8 +31,20 @@ def get_paginated_response(self, url, params={}, **kwargs):
             # params are included in the next url
             params = {}
 
+    def get_identity(self, uuid):
+        '''Returns the details of the identity.'''
+        r = self.session.get(
+            '%s/api/v1/identities/%s/' % (self.base_url, uuid))
+        if r.status_code == 404:
+            return None
+        return r.json()
+
     def get_addresses(self, uuid):
-        '''Get the list of addresses for an identity specified by uuid.'''
+        '''Get the list of addresses that a message to an identity specified
+        by uuid should be sent to.'''
+        identity = self.get_identity(uuid)
+        if identity and identity.get('communicate_through') is not None:
+            return self.get_addresses(identity['communicate_through'])
```

Comment:
Addressed in https://github.com/praekelt/casepro/pull/16/commits/fb8cc17ba1b8aa406779303657753feaae43202c